tests/units/aps/test_fetch_from_api.py

```diff
@@ -0,0 +1,58 @@
+import io
+from datetime import date, datetime
+
+from aps.APSParams import APSParams
+from aps.fetch_from_api import save_file_in_s3, set_APS_harvesting_interval, split_json
+
+today = date.today()
+expected_days = {
+    "aps_fetching_from_date": today.strftime("%Y-%m-%d"),
+    "aps_fetching_until_date": today.strftime("%Y-%m-%d"),
+}
+
+
+class MockedRepo:
+    def find_the_last_uploaded_file_date(self):
+        today = date.today()
+        return today
+
+    def save(self, key, file):
+        pass
+
+    def find_by_id(self, id):
+        return io.BytesIO(
+            str.encode(
+                '{"data":[{"abstract":{"value":"<p>We propose and theoretically analyze</p>"},  "identifiers":{"doi":"10.1103/PhysRevLett.126.153601"}}]}'
+            )
+        )
+
+
+class MockedAPSApiClient:
+    def get_articles_metadata(self, parameters):
+        return str.encode(
+            '{"data":[{"abstract":{"value":"<p>We propose and theoretically analyze</p>"}}]}'
+        )
+
+
+def test_set_APS_harvesting_interval(repo=MockedRepo()):
+    dates = set_APS_harvesting_interval(repo)
+    assert dates == expected_days
+
+
+def test_save_file_in_s3(
+    rest_api=MockedAPSApiClient(),
+    repo=MockedRepo(),
+    parameters=APSParams(expected_days).get_params(),
+):
+    exptected_key = f'{today.strftime("%Y-%m-%d")}/{ datetime.now().strftime("%Y-%m-%dT%H:%M")}.json'
```

Comment:
this test will be flaky, instead you could "hardcode" the dates using https://github.com/spulec/freezegun this will help the test to always run with the same date and time and not rely on `datetime.now()` which will be different in every run